bin/reth/src/db/snapshots/headers.rs

```diff
@@ -23,15 +26,24 @@ impl Command {
         inclusion_filter: InclusionFilter,
         phf: PerfectHashingFunction,
     ) -> eyre::Result<()> {
-        let segment = Headers::new(
-            compression,
-            if self.with_filters {
-                Filters::WithFilters(inclusion_filter, phf)
-            } else {
-                Filters::WithoutFilters
-            },
-        );
-        segment.snapshot::<DB>(provider, self.from..=(self.from + self.block_interval - 1))?;
+        let range = self.from..=(self.from + self.block_interval - 1);
```

Comment:
let's move this range calculation into a separate method to minimize the risk of error